### PR TITLE
Remove Azure Pipelines as a supported CI provider

### PR DESCRIPTION
**Pull request by issyl0**
- Homebrew/linuxbrew-core was the only user of Azure Pipelines, but
  we've switched to GitHub Actions.

### CODE REVIEW

**Comment by issyl0** [comment]
Oh, of course, the CI checks still try to run for Azure.

**Comment by ladislas** [comment]
why remove it? people (including @osx-cross) are using it for their tap process.

**Comment by issyl0** [comment]
I was basing this on the fact we [removed Travis and Circle when they weren't used in any official taps](https://github.com/Homebrew/homebrew-test-bot/pull/312). @MikeMcQuaid What do you think here?

**Comment by MikeMcQuaid** [comment]
@ladislas asked by private email if he could have time to migrate his tap. Have you done that yet @ladislas? Thanks!

File: cmd/test-bot.rb [hunk 1]
```diff
@@ -100,11 +93,10 @@ def setup_argv_and_env
 
     jenkins_pr = !ENV["ghprbPullLink"].nil?
     jenkins_pr ||= !ENV["ROOT_BUILD_CAUSE_GHPRBCAUSE"].nil?
-    azure_pipelines_pr = ENV["BUILD_REASON"] == "PullRequest"
     github_actions_pr = ENV["GITHUB_EVENT_NAME"] == "pull_request"
 
     if ARGV.include?("--ci-auto")
-      if jenkins_pr || azure_pipelines_pr || github_actions_pr
+      if jenkins_pr ||  github_actions_pr
```
Comment (MikeMcQuaid) [inline cmd/test-bot.rb]:
```suggestion
      if jenkins_pr || github_actions_pr
```

File: cmd/test-bot.rb [hunk 2]
```diff
@@ -116,7 +108,7 @@ def setup_argv_and_env
       ARGV << "--cleanup"
       ARGV << "--test-default-formula"
       ARGV << "--local" if jenkins
```
Comment (MikeMcQuaid) [inline cmd/test-bot.rb]:
```suggestion
      ARGV << "--local" << "--junit" if jenkins
```

File: cmd/test-bot.rb [hunk 3]
```diff
@@ -116,7 +108,7 @@ def setup_argv_and_env
...
-      ARGV << "--junit" if jenkins || azure_pipelines
+      ARGV << "--junit" if jenkins
```
Comment (MikeMcQuaid) [inline cmd/test-bot.rb]:
```suggestion
```

**Comment by ladislas** [comment]
Yes you can merge.

**Comment by MikeMcQuaid** [comment]
Thanks for this @issyl0 🎉 ❤️

**Comment by sashkab** [comment]
So tap-new still creates azure-pipelines.yml, while Azure Pipelines no longer supported. Is this intended?

https://github.com/Homebrew/brew/blob/d556d02cd48de62f7f621488405bef5ac74f6998/Library/Homebrew/dev-cmd/tap-new.rb#L52

**Pull request merged**